src/main/java/pl/akademia/api/BikeService.java

```diff
@@ -37,4 +39,22 @@ public int deleteBikeById(Long id) {
   public void switchPrice(BigDecimal value) {
     bikeRepository.switchPrice(value);
   }
+
+  @Transactional
+  public Double averagePriceByBrand(String brand){
+    return bikeRepository.averagePriceByBrand(brand);
+  }
+
+  public Set<String> allBrands(){
+    Set<String> brands = new HashSet<>();
+    for(Bike b : bikeRepository.findAll()){
+      brands.add(b.getBrand());
+    }
+    return brands;
+  }
+
+  @Transactional
```

Comment:
Czy adnotacja @Transactional jest tu wymagana? Jeżeli dobrze rozumiem jej funkcjonalność to generalnie powinna być użyta w przypadku gdy modyfikujemy dane w DB. W tym przypadku tylko je odczytujemy.